### DISCUSSION TITLE
Threshold based on other rules firing

### DISCUSSION THREAD

**Discussion by corycandia** [Security-Onion-Solutions/securityonion#5156] [Unsupported Versions] [answered]
Community,

I'm in the process of tuning a new installation (search master and forward sensor, 2.3.61, ISO install).

I can easily trigger a significant amount of matches of 'ET USER_AGENTS WinRM User Agent Detected - Possible Lateral Movement' using Windows Server Manager tools on a workstation.  

This is normal traffic between workstations and servers, so one could suppress it based on source (if using static addressing on your management workstation).  However, if there was another compromise based detection event (which could be many) on the same workstation prior to the WinRM detection firing, this tells a much different story.  Here's a great article talking about this exact scenario: [https://redcanary.com/blog/lateral-movement-winrm-wmi/](https://redcanary.com/blog/lateral-movement-winrm-wmi/).

In my head, I want to modify the rule or perhaps even better, set a threshold on the event that stipulates another detection of a certain magnitude should take place on the same source host before firing this detection.  I haven't seen anyone write an example of this, or know if it's even easily done without some sort of customization.

I'm not specialized in this, so anyone already know how to do this, have any thoughts, or other solutions?  

(I understand I may be flirting with the whole reason IT security professionals are necessary and get paid well to use their big brains, but really I just don't want to clutter up the in alerts interface with a bunch of stuff I have to check for correlating events for.  That's a bit of a time consumer.)

**Comment by weslambert** [comment, marked as answer]
You could modify rules to check or not check for certain flowbits () having fired, or you could threshold based on various factors, as described here:

https://www.snort.org/faq/readme-thresholding
https://blog.snort.org/2011/05/resolving-flowbit-dependancies.html

The docs pertain to Snort, but are mostly applicable for Suricata as well.
